File: 026/src/Main.java
import us.sodiumlabs.math.prime.PrimeCalculator;

import java.math.BigInteger;

public class Main {
    private PrimeCalculator primeCalculator = new PrimeCalculator();

    private int maxSize = 0;

    public static void main(final String[] args) {
        final Main main = new Main();

        for(int i = 1; i < 1000; i++) {
            main.compareCycleSize(i);
        }
    }

    public void compareCycleSize(final int p) {
        final BigInteger cycle = cyclic(p);
        final int cycleSize = cycle.toString().length();

        if(cycleSize > maxSize) {
            maxSize = cycleSize;
            System.out.println(p + ": " + maxSize + " : " + cycle.toString());
        }
    }

    private BigInteger cyclic(final int p) {
        if( primeCalculator.isPrime(p) && !BigInteger.TEN.gcd(BigInteger.valueOf(p)).equals(BigInteger.TEN) ) {
            return BigInteger.TEN.pow(p).subtract(BigInteger.ONE).divide(BigInteger.valueOf(p));
        }

        return BigInteger.ZERO;
    }
}

File: 026/src/main.js
/**
 * Created by Alex on 4/15/2015.
 */

function cyclic (p) {
    return (Math.pow(10, p - 1) - 1) / p;
}

function main() {
    console.log(cyclic(7));
    console.log(cyclic(17));
    console.log(cyclic(119));
    console.log(cyclic(19));
    console.log(cyclic(61));
}

main();